### DISCUSSION TITLE
Improvment for documentation for usage behind reverse proxy

### DISCUSSION THREAD

**Discussion by hahahannes** [minio/minio#17985] [Q&A] [answered]
I encountered a problem when trying to use Minio behind a reverse proxy. I followed the instructions here: https://min.io/docs/minio/linux/integrations/setup-nginx-proxy-with-minio.html
But it dit not work and I got a 403 for all web socket connections. Especially the settings needed for web sockets in nginx were not enough:
```
proxy_http_version 1.1;
proxy_set_header Upgrade $http_upgrade;
proxy_set_header Connection "upgrade";
```
In my case (Kubernetes & nginx/nginx-ingress:1.8.1), I needed to add the following header to make it work.
```
proxy_set_header Origin '';
```

I am not really sure, whether I misconfigured something or not. Just posting this here, in case someone stumbles upon the 403 on web socket.

**Comment by hahahannes** [comment]
I also noticed that `proxy_set_header Host $http_host` from the documentation cannot work when the Kubernetes NGINX ingress controller is used, as it will always add `proxy_set_header Host $host`

**Comment by harshavardhana** [comment]
@ravindk89 ^^

**Reply by ravindk89** [reply]
:thinking: not sure what the actual fixup here is.

We can add a comment that `proxy_set_header Origin '';` may be required  if using Kubernetes + Nginx ingress.

But the `proxy_set_header` is harder to fix. Does it require overriding the ingress controller? will it always be broken?

Worth noting we had a discussion a while back on the difference between `$http_host` and `$host`, and some users noted that one worked while the other didn't. 

@hahahannes if you have some insight that would help as well - we are not necessarily NGINX experts.

**Comment by ravindk89** [comment, marked as answer]
Updated https://github.com/minio/docs/blob/main/source/integrations/setup-nginx-proxy-with-minio.rst with the note on `Origin`.

I'm also seeing some discussions about using something like `add_header Access-Control-Allow-Origin *` as a wildcard match to direct the client browser to allow CORS from any origin. I don't 100% know how to test for this though, and am unsure if it might open up some security issues.
